src/checkpoint/checkpoint.ts

```diff
@@ -70,32 +63,86 @@ export default class Checkpoint {
 
   public async start() {
     this.log.debug('starting');
+
+    // TODO(perfectmak): Validate config before starting
+
+    await this.store.createStore();
```

Comment:
Any reason to create the tables every time you call `start()`? Shouldn't this happen only once you do a `reset()` to create table if it doesn't exist?

Reply:
I'm of the opinion that for users of Checkpoint, we will want them to only always call `start()` to kick to start (or resume) the indexing. `store.createStore()` only creates the core database tables if they don't exist. So this call effectively acts as a sanity check to ensure core tables are initialized for the first time a user is running the index logic against their database.

I assume `reset()` was meant to allow users to clear their checkpoint store to start all over again, that is why I updated the `reset()` methods implementation to call `store.deleteStore()` before calling `store.createStore()`. So users can choose to call this whenever they want to start all over again (or maybe re-seed their checkpoint blocks).

Let me know if my assumptions are wrong and what was the scenario/usecase you had in mind? @bonustrack

Reply:
@perfectmak Yes the way `reset()` works is that it delete all the table and generate them again, it's a bit similar than doing a deploy on TheGraph. `start()` is just here to continue the process to sync the server. I'm not sure what need to change here, it's not ideal to trigger creation of all table on `start()`, we may need later on to run some query that can't be run twice (like adding default values in the db) maybe we can store the information if the instance already initiated all the db within the db itself on `_metadata` table.

Reply:
Hmm... Okay, this clears things up. I'll try and add comments as well to describe what you've said.

Now, I understand why you say reset() should most likely be called whenever there is a change. And because `reset` will not clear the core `_checkpoints` table, the re-indexing will be faster on start. I'll make the changes and you can review. @bonustrack